include/storage/dedicated_io.hpp

```diff
@@ -0,0 +1,170 @@
+#ifndef OSRM_STORAGE_DEDICATED_IO_HPP_
+#define OSRM_STORAGE_DEDICATED_IO_HPP_
+
+#include "contractor/query_edge.hpp"
+#include "extractor/extractor.hpp"
+#include "extractor/original_edge_data.hpp"
+#include "extractor/query_node.hpp"
+#include "util/exception.hpp"
+#include "util/fingerprint.hpp"
+#include "util/simple_logger.hpp"
+#include "util/static_graph.hpp"
+#include "storage/io.hpp"
+
+#include <boost/filesystem/fstream.hpp>
+#include <boost/iostreams/seek.hpp>
+
+#include <cerrno>
+#include <cstring>
+#include <tuple>
+#include <type_traits>
+
+namespace osrm
+{
+namespace storage
+{
+namespace dedicated_io
```

Comment:
@daniel-j-h `dedicated_io` could have a better name. It's basically the parts of FileReader that is reading a specialized type of input. For example, `readHSGRHeader` which "Reads the checksum, number of nodes and number of edges written in the header file of a `.hsgr` file and returns them in a HSGRHeader struct". This is not your typical filestream that simply opens up a file, verifies a fingerprint or not and returns the FileReader object. 

This file has `readHSGRHeader`, `readPropertiesCount`, `readDatasourceIndexes`, `readEdges`, `readNodes`, `readDatasourceNames` and the assoicated structs. 

It strikes me that these are all related to reading graphs and graph components. So perhaps I can rename this header to "graph_io" or something similar?